src/modules/project/pages/projectView/views/projectMainBody/components/FollowButton.tsx

```diff
@@ -29,6 +36,15 @@ export const FollowButton = ({ project, hasIcon, ...rest }: FollowButtonProps) =
       return
     }
 
+    if (shouldPrompt) {
+      emailPromptOnOpen()
+      return
+    }
+
+    handleFollowUnfollow()
+  }
+
```

Comment:
this function handles both follow and unfollow, So it might trigger if a user is trying to unfollow as well.
 I think only after the follow is done, that we should trigger this.